portalnet/src/storage.rs

```diff
@@ -744,6 +677,496 @@ impl PortalStorage {
+                                }
+                            }
+                        }
+                        _ => {
+                            // Unknown content type
+                            return Err(ContentStoreError::InvalidData {
+                                message: "Unexpected LightClientUpdatesByRange content key"
+                                    .to_string(),
+                            });
+                        }
+                    }
+                }
+            }
+            Some(&LIGHT_CLIENT_FINALITY_UPDATE_KEY_PREFIX) => {
+                // TODO: Store LightClientFinalityUpdate in cache
+            }
+            Some(&LIGHT_CLIENT_OPTIMISTIC_UPDATE_KEY_PREFIX) => {
+                // TODO: Store LightClientOptimisticUpdate in cache
+            }
+            _ => {
+                // Unknown content type
+                return Err(ContentStoreError::InvalidData {
+                    message: "Unknown beacon content key".to_string(),
+                });
+            }
+        }
+
+        let total_bytes_on_disk = self.get_total_storage_usage_in_bytes_on_disk()?;
+        self.metrics
+            .report_total_storage_usage_bytes(total_bytes_on_disk as f64);
+
+        Ok(())
+    }
+
+    pub fn paginate(
+        &self,
+        _offset: &u64,
+        _limit: &u64,
+    ) -> Result<PaginateLocalContentInfo, ContentStoreError> {
+        Err(ContentStoreError::Database(
+            "Paginate not implemented for Beacon storage".to_string(),
+        ))
+    }
+
+    /// Public method for determining how much actual disk space is being used to store this node's Portal Network data.
+    /// Intended for analysis purposes. PortalStorage's capacity decision-making is not based off of this method.
+    pub fn get_total_storage_usage_in_bytes_on_disk(&self) -> Result<u64, ContentStoreError> {
+        let storage_usage = get_total_size_of_directory_in_bytes(&self.node_data_dir)?;
+        Ok(storage_usage)
+    }
+
+    /// Internal method for measuring the total amount of requestable data that the node is storing.
+    fn get_total_storage_usage_in_bytes_from_network(&self) -> Result<u64, ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        let mut query = conn.prepare(TOTAL_DATA_SIZE_QUERY)?;
+
+        let result = query.query_map([], |row| {
+            Ok(DataSize {
+                num_bytes: row.get(0)?,
+            })
+        });
+
+        let sum = match result?.next() {
+            Some(total) => total,
+            None => {
+                let err = "Unable to compute sum over content item sizes".to_string();
+                return Err(ContentStoreError::Database(err));
+            }
+        }?
+        .num_bytes;
+
+        self.metrics.report_content_data_storage_bytes(sum);
+
+        Ok(sum as u64)
+    }
+
+    pub fn lookup_content_key(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
```

Comment:
Am I missing something? Is this not essentially the same as `lookup_content_value()`?

Reply:
`lookup_content_value` accepts content id and give us a content value result, while `lookup_content_key` accepts content id and returns a content key.

Reply:
Ahhh, yeah that makes sense. Imo it's worth adding a docstring here (and to `lookup_content_value()`), even though the fn names are quite descriptive, it can be useful (at least I would've found it useful in this case) to have the functionality spelled out in plain english